### ISSUE TITLE
About prediction result

### ISSUE THREAD

**Issue by aldll** [lyst/lightfm#295]
My input is 10 users and 3000 items,
item features are vectors( tf-idf from text)
the problem is my prediction result is strange,
if user-item interaction is like this:
[[1,0,0,0,0,0, ...],
[0,0,0,0,1,0, ... ],
... the others are all zeros

user A has read item 0 and user B has read item 4.
Item 0 and item 4 are not similar.
However, prediction for user A always shows that item 4's score is the highest. Is there something I'm missing?
Many thanks.

**Comment by maciejkula** [comment]
I may not be able to give you a satisfactory answer without information.

1. Can you run this without item features first?
2. What how many epochs are you running? What are your other hyperparameters?

Finally, this is an unusually small dataset.

**Comment by aldll** [comment]
1. OK, after ran it, I still got the same situation.
2. I have tried epochs=30, 50, 100
hyperparameters from model.get_params():
```{'epsilon': 1e-06,
 'item_alpha': 0.0,
 'k': 5,
 'learning_rate': 0.05,
 'learning_schedule': 'adagrad',
 'loss': 'warp',
 'max_sampled': 10,
 'n': 10,
 'no_components': 10,
 'random_state': <mtrand.RandomState at 0x7f2b7ae5ffc0>,
 'rho': 0.95,
 'user_alpha': 0.0}```

You said 'an unusually small dataset' is that mean the number of user shouldn't too little or something else?
Thanks a lot!

**Comment by maciejkula** [comment]
Could you please post a gist that reproduces your issue (so that I can run it), describing what you _should_ happen and what _is_ happening.

Fitting models on datasets with 10 users is not the use case the package was designed and tested for; it's not clear what's even supposed to happen.

**Comment by aldll** [comment]
I though my data about item features' dimension is too large so that the latent representation calculation(dot and sum) makes something wrong in prediction, it that correct??

And I want to know how user-item interactions influence the prediction, 
what I only know is user features and item features mapping to latent representation and do the dot calculation to get the score. But how user-item interactions do in this process?

**Comment by aldll** [comment]
After some trying, tf-idf normalization has eliminated this situation. Thank you for your patiently replying my question and gave me the advise and opinion about input data!

**Comment by maciejkula** [comment]
Great! Sorry I didn't get to this earlier.
